### PR TITLE
COE-395: Expand planning artifact schema and session service

### PR DESCRIPTION
**Pull request by kumanday**
## Summary

Expands the planning artifact schema and adds the full planning session service models required by the planning wave.

### Planning Artifact Schema (`planning.rs`)
- Added **19 planning artifact kinds**: `intake`, `project_context`, `requirements`, `research_brief`, `codebase_analysis`, `architecture_notes`, `risk_register`, `milestone_draft`, `issue_draft`, `sub_issue_draft`, `dependency_map`, `verification_plan`, `acceptance_criteria`, `plan_validation`, `linear_draft`, `review_comments`, `publish_receipt`, `planning_wave`.
- Added **Display impls** for `PlanningArtifactKind`, `TurnRole`, and `PlanningSessionStatus`, all delegating to `serde_json` serialization so the string output stays in sync with `#[serde(rename_all = "snake_case")]`.
- Added `PlanningSession`, `PlanningSessionSummary`, `ConversationTurn`, `PlanningSessionStatus`, and `ReviewComment` models.
- Added rendering helpers: `render_review_markdown()`, `render_prompt_context()`, `render_audit_history()`.
- Added `PlanningWave`, `TaskEntry`, `TaskPackageProjection` for task-package projections.
- Added `LinearPublishReceipt`, `PublishedMilestone`, `PublishedTask` with `render_yaml()` that uses `serde_yaml` via Serialize-derived view structs for compiler-enforced field coverage and rename consistency.

### Tests (`gateway_schema.rs`)
- Full roundtrip tests for all planning types.
- `render_review_markdown`, `render_prompt_context`, `render_audit_history` content assertions.
- `planning_session_empty_render_is_valid` for empty-session edge case.
- `linear_publish_receipt_render_yaml_is_valid_yaml` parses rendered YAML back and validates structural fields.
- `linear_publish_receipt_render_yaml_is_valid_yaml` asserts exact top-level key set via `HashSet` to catch missing or spurious keys.
- Compile-time gate (`all_planning_types_compile_and_export`) ensures all types are exported.

## Evidence

```
$ cargo test --test gateway_schema
running 42 tests
test result: ok. 42 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

All 42 gateway-schema tests pass, covering every planning type, rendering path, and YAML roundtrip.

## Manual Verification

- `render_yaml()` output parses back through `serde_yaml::from_str()` and matches expected camelCase keys.
- Display output for all enums matches `snake_case` serialization (e.g. `TurnRole::Agent` produces `agent`).
- Empty session renders omit artifact and conversation sections as expected.

### CODE REVIEW

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 1]
```diff
@@ -19,32 +22,148 @@ pub struct PlanningArtifact {
...
+            PlanningArtifactKind::VerificationPlan => write!(f, "verification_plan"),
+            PlanningArtifactKind::AcceptanceCriteria => write!(f, "acceptance_criteria"),
+            PlanningArtifactKind::PlanValidation => write!(f, "plan_validation"),
+            PlanningArtifactKind::LinearDraft => write!(f, "linear_draft"),
+            PlanningArtifactKind::ReviewComments => write!(f, "review_comments"),
+            PlanningArtifactKind::PublishReceipt => write!(f, "publish_receipt"),
+            PlanningArtifactKind::PlanningWave => write!(f, "planning_wave"),
+        }
+    }
 }
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs:75]: Fixed in 3692cef: Display impl now delegates to serde_json serialization, ensuring output stays in sync with the serde rename_all = "snake_case" contract. No manual match arms needed anymore.

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 2]
```diff
@@ -54,3 +173,226 @@ pub enum PlanningSessionStatus {
...
+            lines.push(format!("  {}:", task.task_id));
+            lines.push(format!("    issue: {}", task.issue));
+            lines.push(format!("    issueId: {}", task.issue_id));
+            lines.push(format!("    url: {}", task.url));
+            lines.push(format!("    file: {}", task.file));
+        }
+
+        lines.join("\n")
+    }
+}
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs:443]: Fixed in 3692cef: Replaced hand-built format!() YAML with serde_yaml Mapping builder. Output is now properly escaped and structurally safe.

File: crates/opensymphony-gateway-schema/tests/gateway_schema.rs [hunk 3]
```diff
@@ -449,3 +454,547 @@ fn all_schema_modules_compile_and_export() {
...
+
+#[test]
+fn all_planning_types_compile_and_export() {
+    let _ = PlanningArtifactKind::Intake;
+    let _ = PlanningArtifactKind::ProjectContext;
+    let _ = PlanningArtifactKind::Requirements;
+    let _ = PlanningArtifactKind::ResearchBrief;
+    let _ = PlanningArtifactKind::CodebaseAnalysis;
+    let _ = PlanningArtifactKind::ArchitectureNotes;
+    let _ = PlanningArtifactKind::RiskRegister;
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/tests/gateway_schema.rs:1048]: Fixed in 3692cef: Added new test linear_publish_receipt_render_yaml_is_valid_yaml that parses the rendered output via serde_yaml and verifies all structural fields.

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 4]
```diff
@@ -54,3 +159,260 @@ pub enum PlanningSessionStatus {
...
+            t_mapping.insert("url".into(), serde_yaml::Value::String(task.url.clone()));
+            t_mapping.insert("file".into(), serde_yaml::Value::String(task.file.clone()));
+            tasks.push(serde_yaml::Value::Mapping(t_mapping));
+        }
+        mapping.insert("tasks".into(), serde_yaml::Value::Sequence(tasks));
+
+        serde_yaml::to_string(&mapping)
+            .expect("LinearPublishReceipt yaml serialization should never fail")
+    }
+}
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs:443]: Fixed in 0b6ce58: Replaced manual serde_yaml Mapping builder with Serialize-derived view structs (LinearPublishReceiptYaml, PublishedMilestoneYaml, PublishedTaskYaml). The compiler now enforces field coverage and rename consistency.

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 5]
```diff
@@ -54,3 +159,260 @@ pub enum PlanningSessionStatus {
...
+
+        if let Some(ref wave) = self.planning_wave {
+            parts.push(format!("[Wave: {}]", wave));
+        }
+
+        for artifact in &self.artifacts {
+            parts.push(format!(
+                "[Artifact: {}] {}\n{}",
+                artifact.kind, artifact.title, artifact.content
+            ));
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs:269]: Fixed in 0b6ce58: Added Display impl for PlanningSessionStatus and TurnRole, both delegating to serde_json for consistent snake_case output. Updated render_review_markdown and render_audit_history to use Display ({}) instead of Debug ({:?}).

File: crates/opensymphony-gateway-schema/tests/gateway_schema.rs [hunk 6]
```diff
@@ -449,3 +454,589 @@ fn all_schema_modules_compile_and_export() {
...
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/tests/gateway_schema.rs:1048]: Fixed in 0b6ce58: Added HashSet assertion that top-level YAML keys exactly match the expected set (planningWave, linearProject, publishedAt, milestones, tasks). Any missing or spurious key will now fail the test.

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 7]
```diff
@@ -19,32 +22,144 @@ pub struct PlanningArtifact {
...
+
+impl std::fmt::Display for TurnRole {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Delegate to serde so Display stays in sync with rename_all = "snake_case".
+        match self {
+            TurnRole::User => write!(f, "user"),
+            TurnRole::Agent => write!(f, "agent"),
+            TurnRole::System => write!(f, "system"),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs:137]: Fixed in f5ada0d: Switched TurnRole Display impl to serde_json delegation, matching PlanningArtifactKind. No more manual match arms.

File: crates/opensymphony-gateway-schema/src/planning.rs [hunk 8]
```diff
@@ -54,3 +169,285 @@ pub enum PlanningSessionStatus {
...
+impl std::fmt::Display for PlanningSessionStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Delegate to serde so Display stays in sync with rename_all = "snake_case".
+        match self {
+            PlanningSessionStatus::Draft => write!(f, "draft"),
+            PlanningSessionStatus::InReview => write!(f, "in_review"),
+            PlanningSessionStatus::Approved => write!(f, "approved"),
+            PlanningSessionStatus::Published => write!(f, "published"),
+            PlanningSessionStatus::Archived => write!(f, "archived"),
+        }
```
[the remark this answers is not in the document]
Comment (kumanday) [reply crates/opensymphony-gateway-schema/src/planning.rs]: Fixed in f5ada0d: Switched PlanningSessionStatus Display impl to serde_json delegation, matching PlanningArtifactKind. No more manual match arms.
